index.d.ts

```diff
@@ -0,0 +1,63 @@
+declare namespace tarstream {
+  interface Header {
+    name: string
+    size: number
+    mode: number
+    mtime: Date
+    type: string
+    linkname: string
+    uid: number
+    gid: number
+    uname: string
+    gname: string
+    devmajor: number
+    devminor: number
+    pax?: object
+  }
+
+  type HeaderForPack = Partial<Header> & {
+    name: string
+  }
+
+  interface Pack extends NodeJS.ReadableStream {
+    entry(header: HeaderForPack, callback?: (error?: Error) => void): Sink
+    entry(
+      header: HeaderForPack,
+      buffer: string | Buffer,
```

Comment:
buffer is optional

Reply:
Yes. When buffer is not provided, the signature would be the one on line 23.